crates/extensions_ui/src/extension_suggest.rs

```diff
@@ -0,0 +1,205 @@
+use std::{sync::Arc, hash::{DefaultHasher, Hash, Hasher}};
+
+use editor::{Editor, EditorMode};
+use extension::{ExtensionStore, ExtensionApiResponse, ExtensionIndexEntry};
+use gpui::{AppContext, ViewContext, VisualContext, SharedString};
+use language::Language;
+use workspace::{Workspace, notifications::simple_message_notification};
+use language::Event;
+
+use crate::ExtensionsWithQuery;
+
+
+/// Function to initialize subscriptions to Editor Open and Buffer Language Change; called in the Extensions UI init function
+pub(crate) fn init(cx: &mut AppContext) {
+
+    // Watch when the editor is opened
+    cx.observe_new_views(move |editor: &mut Editor, cx| {
+
+        // Only check if the editor is editing in a programming language
```

Comment:
I think we can remove most of these comments, as they're primarily restating what the code is already doing.

Reply:
Ok!